emcc.py

```diff
@@ -2394,7 +2386,7 @@ def generate_html(target, options, js_target, target_basename,
 %s
           };
           emterpretXHR.send(null);
-''' % (shared.Settings.EMTERPRETIFY_FILE, script.inline)
+''' % (shared.JS.get_subresource_location(shared.Settings.EMTERPRETIFY_FILE), script.inline)
```

Comment:
Hm, well, in that case, I can just copy and paste the new functions into the HTML output and update all the one-off `Module.read*`-like to behave like `Module.read*` (parse data URIs without relying on XHR), in which case there'd be no reason to document the CSP `connect-src` concern. This seemed like more of a hassle when there was a whole other library involved, but not as much now.

That said, `script-src` will be an issue with `SINGLE_FILE` in that kind of scenario no matter how we handle it. Right now, `script-src` (if defined at all) must whitelist `data:`, which I've also documented, but if we parse the data URI to a string then we'll still need either `unsafe-inline` or `unsafe-eval` to be able to execute it. I believe `script-src` is most commonly used to whitelist just `'self'` (the current origin) and maybe specific CDNs, which is fine for regular non-`SINGLE_FILE` usage but would be a problem here.